apps/crn-server/src/handlers/calendar/subscribe-handler.ts

```diff
@@ -1,57 +1,60 @@
 import {
   AlertsSentry,
-  calendarCreatedHandlerFactory,
+  calendarCreatedSquidexHandlerFactory,
+  calendarCreatedContentfulHandlerFactory,
   getJWTCredentialsFactory,
   subscribeToEventChangesFactory,
   unsubscribeFromEventChangesFactory,
 } from '@asap-hub/server-common';
-import { RestCalendar, SquidexGraphql, SquidexRest } from '@asap-hub/squidex';
 import * as Sentry from '@sentry/serverless';
 import 'source-map-support/register';
 import {
-  appName,
   asapApiUrl,
-  baseUrl,
+  contentfulEnvId,
+  contentfulSpaceId,
+  contentfulAccessToken,
   googleApiCredentialsSecretId,
   googleApiToken,
   googleApiUrl,
+  isContentfulEnabledV2,
   region,
 } from '../../config';
-import { CalendarSquidexDataProvider } from '../../data-providers/calendars.data-provider';
-import { getAuthToken } from '../../utils/auth';
+import { getCalendarDataProvider } from '../../dependencies/calendars.dependencies';
 import logger from '../../utils/logger';
 import { sentryWrapper } from '../../utils/sentry-wrapper';
 
-const squidexGraphqlClient = new SquidexGraphql(getAuthToken, {
-  appName,
-  baseUrl,
-});
-const calendarRestClient = new SquidexRest<RestCalendar>(
-  getAuthToken,
-  'calendars',
-  { appName, baseUrl },
-);
-const calendarDataProvider = new CalendarSquidexDataProvider(
-  calendarRestClient,
-  squidexGraphqlClient,
-);
 /* istanbul ignore next */
 const getJWTCredentialsAWS = getJWTCredentialsFactory({
   googleApiCredentialsSecretId,
   region,
 });
-const webhookHandler = calendarCreatedHandlerFactory(
+
+export const calendarCreatedHandlerFactory = isContentfulEnabledV2
+  ? calendarCreatedContentfulHandlerFactory
+  : calendarCreatedSquidexHandlerFactory;
+
+export const contentfulDeliveryApiConfig = {
+  environment: contentfulEnvId,
+  space: contentfulSpaceId,
+  accessToken: contentfulAccessToken,
+};
+
+export const cms = isContentfulEnabledV2 ? 'contentful' : 'squidex';
+
+export const webhookHandler = calendarCreatedHandlerFactory(
   subscribeToEventChangesFactory(getJWTCredentialsAWS, logger, {
     asapApiUrl,
     googleApiToken,
     googleApiUrl,
+    cms,
   }),
   unsubscribeFromEventChangesFactory(getJWTCredentialsAWS, logger, {
     googleApiUrl,
   }),
-  calendarDataProvider,
+  getCalendarDataProvider(),
```

Comment:
is this the suggestion https://github.com/yldio/asap-hub/pull/3046/commits/e23170eabab91618656c4edcb8f0a11d8711a2cd? if not, could you write some pseudo code or some basic description so I can understand the suggestion?

I'm still passing getCalendarDataProvider because I don't want to duplicate the code. `getCalendarDataProvider` is being used in some other places (`resubscribe-handler.ts` and `webhook-events-updated.ts`) so this function still worth existing and think it's kind of weird to duplicate it here in `subscribe-handler.ts` just not to call `isContentfulEnabledV2` under the hood